presto-hive/src/test/java/io/prestosql/plugin/hive/TestShowStats.java

```diff
@@ -212,62 +256,145 @@ public void testShowStatsForNonExistingColumnFails()
+                "SELECT * FROM (VALUES " +
+                        "   ('_col0', null, null, null, null, null, null), " +
+                        "   (null, null, null, null, null, null, null))");
     }
 
     @Test
-    public void testShowStatsWithHavingFails()
+    public void testShowStatsWithHaving()
     {
-        assertQueryFails(
+        assertQuery(
                 "SHOW STATS FOR (SELECT count(nationkey) FROM nation_partitioned GROUP BY regionkey HAVING regionkey > 0)",
-                ".*Only table columns names are supported in SHOW STATS SELECT clause");
+                "SELECT * FROM (VALUES " +
+                        "   ('_col0', null, null, null, null, null, null), " +
+                        "   (null, null, null, null, null, null, null))");
     }
 
     @Test
-    public void testShowStatsWithSelectDistinctFails()
+    public void testShowStatsWithSelectDistinct()
     {
-        assertQueryFails("SHOW STATS FOR (SELECT DISTINCT * FROM orders)", ".*DISTINCT is not supported by SHOW STATS SELECT clause");
+        assertQuery("SHOW STATS FOR (SELECT DISTINCT * FROM orders)",
+                "SELECT * FROM (VALUES " +
+                        "   ('orderkey', null, null, null, null, null, null), " +
+                        "   ('custkey', null, null, null, null, null, null), " +
+                        "   ('orderstatus', null, null, null, null, null, null), " +
+                        "   ('totalprice', null, null, null, null, null, null), " +
+                        "   ('orderdate', null, null, null, null, null, null), " +
+                        "   ('orderpriority', null, null, null, null, null, null), " +
+                        "   ('clerk', null, null, null, null, null, null), " +
+                        "   ('shippriority', null, null, null, null, null, null), " +
+                        "   ('comment', null, null, null, null, null, null), " +
+                        "   (null, null, null, null, null, null, null))");
     }
 
     @Test
-    public void testShowStatsWithSelectFunctionCallFails()
+    public void testShowStatsWithSelectFunctionCall()
     {
-        assertQueryFails("SHOW STATS FOR (SELECT sin(orderkey) FROM orders)", ".*Only table columns names are supported in SHOW STATS SELECT clause");
-        assertQueryFails("SHOW STATS FOR (SELECT count(*) FROM orders)", ".*Only table columns names are supported in SHOW STATS SELECT clause");
+        assertQuery("SHOW STATS FOR (SELECT sin(orderkey) FROM orders)",
+                "SELECT * FROM (VALUES " +
+                        "   ('_col0', null, null, null, null, null, null), " +
+                        "   (null, null, null, null, 15000, null, null))");
+        assertQuery("SHOW STATS FOR (SELECT count(*) FROM orders)",
+                "SELECT * FROM (VALUES " +
+                        "   ('_col0', null, null, null, null, null, null), " +
+                        "   (null, null, null, null, null, null, null))");
```

Comment:
row count should be 1, since this is a global agg

this may be useful for "helper" tables used with `max` to pull some information used in oiher places in a query
please add a todo